javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -1355,18 +1349,12 @@ private Natural() {
             public int compare(Tuple2<K, V> e1, Tuple2<K, V> e2) {
                 final K key1 = e1._1;
                 final K key2 = e2._1;
-                if (key1 == null) {
-                    return (key2 == null) ? 0 : -1;
-                } else if (key2 == null) {
-                    return 1;
-                } else {
-                    return ((Comparable<K>) key1).compareTo(key2);
-                }
+                return ((Comparable<K>) key1).compareTo(key2);
```

Comment:
No, this special-handling pollutes our code-base. It is not only the TreeMap - most sorted collections are affected. I added javadoc to Set describing the problem.

Where do we stop adding special *magic* behavior. Has traversable min() use a natural comparator that returns 'null' as minimum. I don't think so.

We do it straight forward - the simplest solution possible.